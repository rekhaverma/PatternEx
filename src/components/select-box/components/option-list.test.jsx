import React from 'react';
import OptionList from './option-list.component';

const optionsList = [
  {
    content: 'All values',
    label: 'All Values',
    id: 'all',
  },
  {
    content: 'Value1',
    label: 'Value1',
    id: 'value1',
  },
  {
    content: 'Value2',
    label: 'Value2',
    id: 'value2',
  },
];

const styleList = {
  marginRight: 20,
};

describe('<OptionList />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      activeOption: 'all',
      className: 'optionList__test',
      onClick: jest.fn(),
      options: optionsList,
      scrollBar: true,
      singleSelect: true,
      style: styleList,
    };
    wrapper = shallow(<OptionList {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

