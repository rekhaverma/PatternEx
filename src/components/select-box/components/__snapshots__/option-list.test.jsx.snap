// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionList /> should render correctly 1`] = `
<div
  className="optionList__test"
>
  <Option
    className="selectBox__option"
    content="All values"
    disabled={false}
    id="all"
    isActive={true}
    key="all"
    label="All Values"
    onClick={[Function]}
    singleSelect={true}
    style={
      Object {
        "marginRight": 20,
      }
    }
  />
  <Option
    className="selectBox__option"
    content="Value1"
    disabled={false}
    id="value1"
    isActive={false}
    key="value1"
    label="Value1"
    onClick={[Function]}
    singleSelect={true}
    style={
      Object {
        "marginRight": 20,
      }
    }
  />
  <Option
    className="selectBox__option"
    content="Value2"
    disabled={false}
    id="value2"
    isActive={false}
    key="value2"
    label="Value2"
    onClick={[Function]}
    singleSelect={true}
    style={
      Object {
        "marginRight": 20,
      }
    }
  />
</div>
`;
